examples/jsm/loaders/PCDLoader.js

```diff
@@ -386,14 +409,15 @@ class PCDLoader extends Loader {
 		if ( position.length > 0 ) geometry.setAttribute( 'position', new Float32BufferAttribute( position, 3 ) );
 		if ( normal.length > 0 ) geometry.setAttribute( 'normal', new Float32BufferAttribute( normal, 3 ) );
 		if ( color.length > 0 ) geometry.setAttribute( 'color', new Float32BufferAttribute( color, 3 ) );
+		if ( intensity.length > 0 ) geometry.setAttribute( 'color', new Float32BufferAttribute( intensity, 3 ) );
```

Comment:
> I would add an intensity attribute, not a color attribute.

An `intensity` attribute can't be used for vertex colors. It's important that user don't have to write custom shaders for this use case.

> I also would not scale it by 255.

The color values have to be in the range [0,1] when using a float color attribute.

Reply:
You are assuming the intensity is used as a replacement for color information. It is a value returned from LiDAR.

Just load the data.

Reply:
It seems to be the common usage to treat intensity as color information (from https://github.com/isl-org/Open3D/issues/125):

> If you use point cloud with intensity, just copy the intensity value to all the rgb fields

Reply:
Also from https://www.mathworks.com/help/vision/ref/pcread.html

> Intensity property, stores the grayscale intensity for each point.

Reply:
LiDAR data is a special-use-case for PCD file format. PCD file format supports optional fields.

Intensity is an optional field, and I think intensity for LiDAR is nominally normalized to 1 or 100 -- not to 255 -- because it is not a color.

I do think this could be a great example -- just hardwiring it into the loader as a color doesn't seem right to me.

JMHO, though. :-)

Reply:
> because it is not a color.

Do you have a resource where this is documented? While researching the topic, I've only found that `intensity` is more a less an alternative to a RGB definition (XYZRGB vs XYZI). This also explains its range between 0 and 255.

Reply:
It's a good idea to add the supported PCD fields to the documentation. I'm fine with the suggested range however all files I have tested with represent the intensity as float values. The first data line of a respective PCD looks like so:
```
# .PCD v0.7 - Point Cloud Data file format
VERSION 0.7
FIELDS x y z intensity
SIZE 4 4 4 4
TYPE F F F F
COUNT 1 1 1 1
WIDTH 7816
HEIGHT 1
VIEWPOINT 0 0 0 1 0 0 0
POINTS 7816
DATA ascii
-0.096134186 -408.10934 13.66958 5.8790812
```
The files actually define `F` (and not `I`) as the data type of `intensity`.

I think I would not add any data type restriction to the docs. The range of intensity and the fact that the loader interprets the data as a grayscale color value should be sufficient.

Reply:
Oh, I meant whole-number, not integer format. Forget I mentioned that topic. :-)

The intensity is a float. I still think it is most reasonable to assume the value is in [0, 100] -- not [0, 255]. That is what is bugging me...

Reply:
I've changed my mind after reading this thread: https://answers.ros.org/question/232118/rviz-point-cloud-intensity-channel-how-does-it-relate-to-point-cloud-xyzi/

Relevant part:

> Intensity values don't have a unit and are device-dependent; that means the actual minimum and maximum depends on the device that produced them, and you cannot directly compare intensities from different sensor models. The only convention is that "higher value" means "more intense".

That means there is no reliable way to scale the intensity to [0,1]. It's just not right to rely on [0, 255] or [0, 100]. 

Let's continue the discussion in #24253.

Reply:
@WestLangley Thanks for not prematurely approving the PR^^.